### ISSUE TITLE
VSC_ARCH_LOCAL validation 

### ISSUE THREAD

**Issue by Lewih** [vscentrum/vsc-test-suite#11]
Can be done using `archspec` which is installed automatically with reframe.
Use `reframe --detect-host-topology` to obtain information about the system.

**Comment by stevenvdb** [comment]
Small remark: `reframe --detect-host-topology` reports `skylake_avx512` on our Genius cluster as architecture name, but `$VSC_ARCH_LOCAL` is set to `skylake`.

**Comment by boegel** [comment]
@stevenvdb Technically speaking, `skylake_avx512` is what's correct (because there's also a consumer CPU nicknamed `Skylake` which does not support AVX-512 instructions).

That said: we have the same in Ghent: we use `skylake`, while `archspec` reports `skylake_avx512`.

As discussed during the last meeting, there's also `zen2` (the technically correct name for the microarchitecture of AMD Rome CPUs) vs the perhaps more sensible `rome` (and likewise for `zen3` vs `milan`). In Ghent, we stick to `zen2` and `zen3`, but there's definitely an argument for `rome` and `milan` instead, since that's closer to code names of Intel CPUs.

Long story short: we can't just compare against what `archspec` produces, at the very least we'll need exceptions.

Going forward, maybe we *can* agree to stick the "official" name that `archspec` reports however, and then gradually phase out the exceptions?

**Comment by Lewih** [comment]
> Long story short: we can't just compare against what archspec produces, at the very least we'll need exceptions.

Let's move in this direction: to relax the test during this iteration I suggest to use regex for cases like `skylake_avx512`. 

> As discussed during the last meeting, there's also zen2 (the technically correct name for the microarchitecture of AMD Rome CPUs) vs the perhaps more sensible rome (and likewise for zen3 vs milan). In Ghent, we stick to zen2 and zen3, but there's definitely an argument for rome and milan instead, since that's closer to code names of Intel CPUs.

We need to handle these exceptions as well: we should have a dictionary containing the possibile matches for an `archspec` output of interest. If not present go straight with output.

#22 has to be slightly modified to add this relaxation.

**Comment by smoors** [comment]
I've relaxed the spec in #22 to allow both the full archspec name and with the `_xxx` part removed.
